istio/istio1106/xds/client.go

```diff
@@ -148,26 +147,27 @@ func (ads *AdsStreamClient) HandleResponse(resp interface{}) {
 	 *              2020-12-01T09:17:29.354132Z     info    ads     Timeout writing sidecar~10.49.18.38~no-project-aabb-gz01a-blue-67cb764fcb-8dq4t.dmall-inner~dmall-inner.svc.cluster.local-39
 	 * So, will use async
 	 */
-	utils.GoWithRecover(func() {
-		var err error
-		// Ads Handler, overwrite the AdsStreamClient to extends more ads or handle
-		switch dresp.TypeUrl {
-		case EnvoyCluster:
-			err = ads.handleCds(dresp)
-		case EnvoyEndpoint:
-			err = ads.handleEds(dresp)
-		case EnvoyListener:
-			err = ads.handleLds(dresp)
-		case EnvoyRoute:
-			err = ads.handleRds(dresp)
-		default:
-			err = errors.New("unsupport type url")
-		}
-		if err != nil {
-			log.DefaultLogger.Errorf("handle typeurl %s failed, error: %v", dresp.TypeUrl, err)
-		}
+	//TODO Concurrency issues
+	//utils.GoWithRecover(func() {
```

Comment:
delete this comment.

Reply:
// TODO: optimise the efficiency of xDS. 
// If xDS resource too big, Istio maybe have written timeout error when use sync, such as:
// 2020-12-01T09:17:29.354132Z     info    ads     Timeout writing sidecar~10.49.18.38